runtime/resolver.go

```diff
@@ -304,6 +307,34 @@ func (r *driverResolverResult) MarshalJSON() ([]byte, error) {
 	return json.Marshal(out)
 }
 
+// Meta implements ResolverResult.
+func (r *driverResolverResult) Meta() []map[string]any {
+	if r.meta == nil {
+		return nil
+	}
+
+	// Build a set of schema field names
+	fieldNames := make(map[string]struct{})
+	if r.rows != nil && r.rows.Schema != nil {
+		for _, f := range r.rows.Schema.Fields {
+			fieldNames[f.Name] = struct{}{}
+		}
+	}
+
+	// Filter meta to only include entries for fields present in the schema
+	var filtered []map[string]any
+	for _, m := range r.meta {
+		name, ok := m["name"].(string)
+		if !ok {
+			continue
+		}
+		if _, exists := fieldNames[name]; exists {
+			filtered = append(filtered, m)
+		}
+	}
+	return filtered
+}
```

Comment:
Not sure I follow this, seems like this is some logic specific to the metrics resolver? It should just return `r.meta` I think, and any logic to transform/enrich the `meta` object should be done in the specific resolver implementation before calling `NewDriverResolverResult`.